sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -215,12 +215,7 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
 
     val optimized = Optimize.execute(query)
 
-    val expected = nestedRelation
-      .select(GetStructField('a, 0, Some("b")))
-      .limit(5)
-      .analyze
-
-    comparePlans(optimized, expected)
+    comparePlans(optimized, query)
```

Comment:
Since we need the whole structure, why we did expect the local relation to be column pruned?

Reply:
Yea, it seems that's just a mistake. cc: @dongjoon-hyun

Reply:
btw, can you add tests in this suite, too?

Reply:
I could add something similar to the SQL test to here as well:
```
  test("SPARK-30870: Don't alias nested field if it means a whole attribute") {
    val valueStructType = StructType.fromDDL("field struct<a:int, b:int>")
    val r = LocalRelation('value.struct(valueStructType))

    val field = GetStructField('value, 0, Some("field"))

    val query = r
      .limit(5)
      .select(field)
      .analyze

    val optimized = Optimize.execute(query)

    comparePlans(optimized, query)
  }
```
but it wouldn't be much different to this particular test (`Some nested column means the whole structure`).

Reply:
Hi, @peter-toth and @maropu .
The original code is correct, this is not about column pruning. This is about `limit` push down.

Reply:
But the test name is `Some nested column means the whole structure`?

Reply:
Ya. It might be misleading now, but It was `true` in the original context.
- https://github.com/apache/spark/pull/23964

Reply:
@dongjoon-hyun hmm, does this test have anything to do with limit push down? There is no `LimitPushDown` is the optimizer of this suite: https://github.com/apache/spark/blob/5a51b9472f5dfbf99ef6f8d6c7151618157c446e/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala#L34-L39 and actually `limit` is closer to the relation in the original `query` than in `expected`, but I might be wrong.

Reply:
Sorry, I meant a pushdown over `limit`.

Reply:
`[SPARK-26975][SQL] Support nested-column pruning over limit/sample/repartition` is about that.

Reply:
Hmm. I got it. So, this is the result of bug fix, isn't it?

Reply:
Yes, it is. In this test case there is no point in pushing down the `project` over the `limit`.